Userland/Libraries/LibJS/Runtime/DisposableStackPrototype.cpp

```diff
@@ -0,0 +1,204 @@
+    define_native_function(realm, vm.names.move, move_, 0, attr);
+
+    // 11.3.3.7 DisposableStack.prototype [ @@dispose ] (), https://tc39.es/proposal-explicit-resource-management/#sec-disposablestack.prototype-@@dispose
+    define_direct_property(*vm.well_known_symbol_dispose(), get_without_side_effects(vm.names.dispose), attr);
+
+    // 11.3.3.8 DisposableStack.prototype [ @@toStringTag ], https://tc39.es/proposal-explicit-resource-management/#sec-disposablestack.prototype-@@toStringTag
+    define_direct_property(*vm.well_known_symbol_to_string_tag(), PrimitiveString::create(vm, vm.names.DisposableStack.as_string()), Attribute::Configurable);
+}
+
+// 11.3.3.1 get DisposableStack.prototype.disposed, https://tc39.es/proposal-explicit-resource-management/#sec-get-disposablestack.prototype.disposed
+JS_DEFINE_NATIVE_FUNCTION(DisposableStackPrototype::disposed_getter)
+{
+    // 1. Let disposableStack be the this value.
+    // 2. Perform ? RequireInternalSlot(disposableStack, [[DisposableState]]).
+    auto* disposable_stack = TRY(typed_this_object(vm));
+
+    // 3. If disposableStack.[[DisposableState]] is disposed, return true.
+    if (disposable_stack->disposable_state() == DisposableStack::DisposableState::Disposed)
+        return Value(true);
+
+    // 4. Otherwise, return false.
+    return Value(false);
+}
+
+// 11.3.3.2 DisposableStack.prototype.dispose (), https://tc39.es/proposal-explicit-resource-management/#sec-disposablestack.prototype.dispose
+JS_DEFINE_NATIVE_FUNCTION(DisposableStackPrototype::dispose)
+{
+    // 1. Let disposableStack be the this value.
+    // 2. Perform ? RequireInternalSlot(disposableStack, [[DisposableState]]).
+    auto* disposable_stack = TRY(typed_this_object(vm));
+
+    // 3. If disposableStack.[[DisposableState]] is disposed, return undefined.
+    if (disposable_stack->disposable_state() == DisposableStack::DisposableState::Disposed)
+        return js_undefined();
+
+    // 4. Set disposableStack.[[DisposableState]] to disposed.
+    disposable_stack->set_disposed();
+
+    // 5. Return DisposeResources(disposableStack, NormalCompletion(undefined)).
+    return TRY(dispose_resources(vm, &disposable_stack->disposable_resource_stack(), Completion { js_undefined() }));
+}
+
+// 11.3.3.3 DisposableStack.prototype.use( value ), https://tc39.es/proposal-explicit-resource-management/#sec-disposablestack.prototype.use
+JS_DEFINE_NATIVE_FUNCTION(DisposableStackPrototype::use)
+{
+    // 1. Let disposableStack be the this value.
+    // 2. Perform ? RequireInternalSlot(disposableStack, [[DisposableState]]).
+    auto* disposable_stack = TRY(typed_this_object(vm));
+
+    // 3. If disposableStack.[[DisposableState]] is disposed, throw a ReferenceError exception.
+    if (disposable_stack->disposable_state() == DisposableStack::DisposableState::Disposed)
+        return vm.throw_completion<ReferenceError>(ErrorType::DisposableStackAlreadyDisposed);
+
+    auto value = vm.argument(0);
```

Comment:
let's put this before step 1 for readability, it's only not needed in an error case anyway